### ISSUE TITLE
Doesn't build with Clang

### ISSUE THREAD

**Issue by lpgauth** [thekvs/erlang-murmurhash#1]
```
erlang-murmurhash (master) $ rebar compile
==> erlang-murmurhash (compile)
Compiled src/erlang_murmurhash.erl
Compiling c_src/MurmurHash2.cpp
Compiling c_src/MurmurHash3.cpp
Compiling c_src/erlang_murmurhash_nif.cpp
c_src/erlang_murmurhash_nif.cpp:119:10: error: no matching function for call to 'enif_get_ulong'
    if (!enif_get_uint64(env, argv[1], &seed)) {
         ^~~~~~~~~~~~~~~
/usr/local/lib/erlang/erts-6.1/include/erl_nif_api_funcs.h:542:27: note: expanded from macro 'enif_get_uint64'
#  define enif_get_uint64 enif_get_ulong
                          ^~~~~~~~~~~~~~
/usr/local/lib/erlang/erts-6.1/include/erl_nif_api_funcs.h:43:27: note: candidate function not viable: no known conversion from 'uint64_t *' (aka 'unsigned long long *') to 'unsigned long *' for 3rd argument
ERL_NIF_API_FUNC_DECL(int,enif_get_ulong,(ErlNifEnv*, ERL_NIF_TERM term, unsigned long* ip));
                          ^
/usr/local/lib/erlang/erts-6.1/include/erl_nif.h:223:71: note: expanded from macro 'ERL_NIF_API_FUNC_DECL'
#  define ERL_NIF_API_FUNC_DECL(RET_TYPE, NAME, ARGS) extern RET_TYPE NAME ARGS
                                                                      ^
c_src/erlang_murmurhash_nif.cpp:154:10: error: no matching function for call to 'enif_get_ulong'
    if (!enif_get_uint64(env, argv[1], &seed)) {
         ^~~~~~~~~~~~~~~
/usr/local/lib/erlang/erts-6.1/include/erl_nif_api_funcs.h:542:27: note: expanded from macro 'enif_get_uint64'
#  define enif_get_uint64 enif_get_ulong
                          ^~~~~~~~~~~~~~
/usr/local/lib/erlang/erts-6.1/include/erl_nif_api_funcs.h:43:27: note: candidate function not viable: no known conversion from 'uint64_t *' (aka 'unsigned long long *') to 'unsigned long *' for 3rd argument
ERL_NIF_API_FUNC_DECL(int,enif_get_ulong,(ErlNifEnv*, ERL_NIF_TERM term, unsigned long* ip));
                          ^
/usr/local/lib/erlang/erts-6.1/include/erl_nif.h:223:71: note: expanded from macro 'ERL_NIF_API_FUNC_DECL'
#  define ERL_NIF_API_FUNC_DECL(RET_TYPE, NAME, ARGS) extern RET_TYPE NAME ARGS
                                                                      ^
2 errors generated.
ERROR: compile failed while processing /Users/lpgauth/Git/erlang-murmurhash: rebar_abort
```

version information:
- Erlang:
  
  erl +V
  Erlang (SMP,ASYNC_THREADS,HIPE) (BEAM) emulator version 6.1
- Clang:
  
  gcc -v
  Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
  Apple LLVM version 5.1 (clang-503.0.40) (based on LLVM 3.4svn)
  Target: x86_64-apple-darwin13.3.0
  Thread model: posix

**Comment by thekvs** [comment]
On my Ubuntu box I've compiled with clang with no problems:

```
kvs@ubuntu-v erlang-murmurhash:> CC=clang CXX=clang++ ./rebar compile
==> erlang-murmurhash (compile)
Compiled src/erlang_murmurhash.erl
Compiling c_src/MurmurHash2.cpp
Compiling c_src/MurmurHash3.cpp
Compiling c_src/erlang_murmurhash_nif.cpp
```

Unfortunately now I don't have access to MacOS, so patches are welcome!

**Comment by sergium** [comment]
the same problem

```
gcc -v
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 6.1.0 (clang-602.0.49) (based on LLVM 3.6.0svn)
Target: x86_64-apple-darwin14.3.0
Thread model: posix
```

**Comment by toland** [comment]
This problem is caused by the fact that `uint64_t` is defined as `unsigned long long` on OSX but `enif_get_ulong`/`enif_get_uint64` is declared to take `unsigned long *`:

``` C
ERL_NIF_API_FUNC_DECL(int,enif_get_ulong,(ErlNifEnv*, ERL_NIF_TERM term, unsigned long* ip));
```

To me, it looks like the simple fix is to use `unsigned long *` instead of `uint64_t` since that is what the function is declared to take. I will prepare a PR with that change.

**Comment by thekvs** [comment]
Fixed in #9
